modules/backup-s3/client.go

```diff
@@ -102,41 +141,30 @@ func (s *s3Client) GetObject(ctx context.Context, backupID, key string) ([]byte,
 	return contents, nil
 }
 
-func (s *s3Client) PutFile(ctx context.Context, backupID, key string, srcPath string) error {
-	objectName := s.makeObjectName(backupID, key)
-	srcPath = path.Join(s.dataPath, srcPath)
-	opt := minio.PutObjectOptions{ContentType: "application/octet-stream"}
-
-	_, err := s.client.FPutObject(ctx, s.config.Bucket, objectName, srcPath, opt)
-	if err != nil {
-		return backup.NewErrInternal(
-			errors.Wrapf(err, "put file '%s'", objectName))
-	}
-
-	// Get filesize
-	file, err := os.Stat(srcPath)
+func (s *s3Client) PutObject(ctx context.Context, backupID, key, overrideBucket, overridePath string, byes []byte) error {
+	client, err := s.getClient(ctx)
 	if err != nil {
-		return nil
+		return errors.Wrap(err, "failed to create temporary client")
 	}
-	size := file.Size()
-
-	metric, err := monitoring.GetMetrics().BackupStoreDataTransferred.GetMetricWithLabelValues(Name, "class")
-	if err == nil {
-		metric.Add(float64(size))
-	}
-	return nil
-}
 
-func (s *s3Client) PutObject(ctx context.Context, backupID, key string, byes []byte) error {
-	objectName := s.makeObjectName(backupID, key)
+	remotePath := s.makeObjectName(backupID, key)
 	opt := minio.PutObjectOptions{ContentType: "application/octet-stream"}
 	reader := bytes.NewReader(byes)
 	objectSize := int64(len(byes))
 
-	_, err := s.client.PutObject(ctx, s.config.Bucket, objectName, reader, objectSize, opt)
+	if overridePath != "" {
+		remotePath = path.Join(overridePath, backupID, key)
+	}
+
+	bucket := s.config.Bucket
+	if overrideBucket != "" {
+		bucket = overrideBucket
+	}
+
+	_, err = client.PutObject(ctx, bucket, remotePath, reader, objectSize, opt)
 	if err != nil {
 		return backup.NewErrInternal(
-			errors.Wrapf(err, "put object '%s'", objectName))
+			errors.Wrapf(err, "put object %s:%s", bucket, remotePath))
```

Comment:
let's add a colon in a message after `put object`

Reply:
done

Reply:
done